### DISCUSSION TITLE
Foreground gradient

### DISCUSSION THREAD

**Discussion by plushrib** [polybar/polybar#3143] [Q&A]
Hello! Is it I possible to make the foreground text a gradient the same way you can make the background a gradient?

**Comment by patrick96** [comment]
No, there is no such feature.

Progressbars have support for a kind of gradient, but that's very restricted; you can't really have arbitrary text and there is no interpolation between colors, only the colors in the configuration are used

**Reply by plushrib** [reply]
what if you write a .sh file and import that as foreground in config?

**Reply by patrick96** [reply]
I'm not sure what you mean by that.

If you have a script module you could simulate gradients (at least horizontal ones) by changing the foreground between characters. But you're still not getting a true gradient within a character

**Comment by plushrib** [comment]
what about editing the renderer.cpp file?

**Reply by patrick96** [reply]
I mean sure, you could go ahead and implement foreground gradients as a feature. Don't know how much effort that would be though

**Reply by plushrib** [reply]
Figured out how to compile that without any errors but it only shows the hex color for "foreground-0" and dosent makes to the text
